device.go

```diff
@@ -203,12 +216,9 @@ func (d *Device) SetAutoDetach(autodetach bool) error {
 	if d.handle == nil {
 		return fmt.Errorf("SetAutoDetach(%v) called on %s after Close", autodetach, d)
 	}
-	var autodetachInt int
-	switch autodetach {
-	case true:
-		autodetachInt = 1
-	case false:
-		autodetachInt = 0
+	d.autodetach = autodetach
+	if autodetach && d.detached == nil{
+		d.detached = make(map[int]bool)
 	}
-	return libusb.setAutoDetach(d.handle, autodetachInt)
```

Comment:
we still need autodetach enabled in libusb - after the device is switched to a new configuration, the kernel may reattach a driver. So you'd need detach before claim again.